lib/components/footer.dart

```diff
@@ -0,0 +1,145 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_web/responsive.dart';
+
+import '../constants.dart';
+
+class Footer extends StatelessWidget {
+  const Footer({
+    Key key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return (!isMobile(context)) ? DesktopFooter() : MobileFooter();
```

Comment:
avoid inverting your conditionals: use 
`condition? trueBranch : falseBranch`
 instead of 
`!condition ? falseBranch : trueBranch`